Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/photo_field.tsx

```diff
@@ -0,0 +1,653 @@
+    })();
+    return (
+      <div>
+        <div style={ChangePictureModal.STYLE.transparentBackground}/>
+        <div style={boxShadowStyle}/>
+        <HBoxLayout style={boxStyle}>
+          <Padding size={ChangePictureModal.PADDING}/>
+          <VBoxLayout>
+            <Padding size={ChangePictureModal.PADDING}/>
+            <div style={ChangePictureModal.STYLE.header}>
+              {ChangePictureModal.HEADER_TEXT}
+              <img src='resources/close.svg'
+                style={ChangePictureModal.STYLE.closeIcon}
+                onClick={this.onClose}/>
+            </div>
+            <Padding size={ChangePictureModal.PADDING_BETWEEN_ELEMENTS}/>
+            <div style={imageBoxStyle}>
+              <img src={imageSrc}
+                style={{ ...imageStyle, ...imageScaling }}/>
+            </div>
+            <Padding size={ChangePictureModal.PADDING_BETWEEN_ELEMENTS}/>
+            <Slider onChangeValue={this.onSliderMovement}
+              scaleValue={this.state.imageScaling}
+              isReadOnly={isSliderReadOnly}/>
+            <Padding size={ChangePictureModal.PADDING_BETWEEN_ELEMENTS}/>
+            <HLine color='#E6E6E6' height={1}/>
+            <Padding size={ChangePictureModal.PADDING_BETWEEN_ELEMENTS}/>
+            <div style={buttonBox}>
+              <input type='file' id='imageInput' accept='image/*'
+                style={ChangePictureModal.STYLE.hiddenInput}
+                onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                  this.onGetImageFile(event.target.files);}}/>
+              <label htmlFor='imageInput'
+                className={css(ChangePictureModal.SPECIAL_STYLE.button)}>
+                {ChangePictureModal.BROWSE_BUTTON_TEXT}
+              </label>
+              <div className={css(ChangePictureModal.SPECIAL_STYLE.button)}
+                onClick={this.onSubmit}>
+                {ChangePictureModal.SUBMIT_BUTTON_TEXT}
+              </div>
+            </div>
+            <Padding size={ChangePictureModal.PADDING}/>
+          </VBoxLayout>
+          <Padding size={ChangePictureModal.PADDING}/>
+        </HBoxLayout>
+      </div>);
+  }
+
+  private onSliderMovement(value: number) {
+    this.setState({ imageScaling: value });
+  }
+
+  private onGetImageFile(selectorFiles: FileList) {
+    const file = selectorFiles.item(0);
+    const someURL = URL.createObjectURL(file);
+    this.setState({
+      currentImage: someURL,
+      imageScaling: 0
+      });
+  }
+
+  private onClose() {
+    this.props.onCloseModal();
+    this.setState({ imageScaling: 0 });
+    this.setState({ currentImage: this.props.imageSource });
+  }
+
+  private onSubmit() {
+    if(this.state.currentImage) {
+      this.props.onSubmitImage(this.state.currentImage,
+        this.state.imageScaling);
+    }
+    this.props.onCloseModal();
+  }
+
+  private static readonly STYLE = {
+    transparentBackground: {
+      boxSizing: 'border-box' as 'border-box',
+      top: '0',
```

Comment:
Be consistent with whether you wish to use a string or a number for CSS. For example, ```top``` uses a ```string``` but ```zIndex``` uses a number. My preference is to use a string with units ie. ```0px``` or ```0%```, and a number in situations where units are not available, like for ```zIndex```. So in this case that means ```top``` should be ```0px``` or ```0%```.